examples/caustics.dx

```diff
@@ -0,0 +1,475 @@
+'Caustic Rendering
+
+Based on https://benedikt-bitterli.me/tantalum/
+
+'## Rendering utilities
+
+AbsPoint = (Int & Int)
+RelPoint = (Float & Float)
+
+def pixel (w:Int)?-> (h:Int)?-> (x:Int) (y:Int) : (Fin w & Fin h) = (x@_, y@_)
+
+def drawIntoCanvas (height:Int) (width:Int)
+                   (f: ref:Type ?-> Ref ref ((Fin width & Fin height) =>out)
+                                ->{Accum ref} Unit)
+                   : Fin height=>Fin width=>out =
+  canvas = snd $ withAccum f
+  for y x. canvas.(x, y)
+
+def checkerboard (height:Int)?-> (width:Int)?-> (ref:Type)?->
+      (canvas:Ref ref ((Fin width & Fin height)=>Float))
+      :{Accum ref} Unit =
+  h_over_two = height `idiv` 2
+  -- note: this works but can't be reduced so it has to be a def (?)
+  for i:(Fin width) j:(Fin h_over_two).
+    canvas!(pixel (ordinal i) ((2 * ordinal j) + (ordinal i `rem` 2))) += 1.0
```

Comment:
does that necessarily make the same checkerboard pattern? I think it would work for odd-sized checkerboards but fail for even-sized ones. (but this isn't really necessary for anything later anyway, so we could just make it only work for odd sizes, I guess? or remove it, if we don't think it adds anything)

Reply:
I mean, I honestly don't understand what's happening in those examples because the comments are very limited. So I don't really have an opinion about keeping or deleting them 😕